### PR TITLE
Update `salesforce` jsdocs 

### LINKED ISSUE #664: salesforce: docs deep dive
This issue is to do a deep docs review on Salesforce. I expect to contribute heavily to this myself.
good examples

Requirements:

* state mutation JSdocs
* careful consistent language
* http helpers must explicitly clearly target salesforce
* update, create and upsert example supports multiple records
* bulk upsert example
* Example for each supported bulk operation
* Ensure examples are clear, concise, and demonstrate best practices.

### PR DESCRIPTION
**Pull request by mtuchi**
## Summary

Update params docs and improve examples in salesforce adaptor  

Fixes #664 

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [ ] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [ ] Strategy / design
- [ ] Optimisation / refactoring
- [x] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

## Review Checklist

Before merging, the reviewer should check the following items:

- [x] Does the PR do what it claims to do?
- [ ] If this is a new adaptor, added the adaptor on marketing website ?
- [ ] Are there any unit tests?
- [x] Is there a changeset associated with this PR? Should there be? Note that
      dev only changes don't need a changeset.
- [x] Have you ticked a box under AI Usage?

### CODE REVIEW

**Comment by josephjclark** [comment]
@mtuchi Please make angry noises at me if this isn't done tomorrow morning

**Comment by josephjclark** [comment]
@mtuchi is this targeted at main? I expected this to be part of the next salesforce stuff, branching from `epic/salesforce`

File: packages/salesforce/src/Adaptor.js [hunk 1]
```diff
@@ -298,21 +320,19 @@ const defaultOptions = {
...
- * <caption>The results will be available on `state.data`</caption>
- * bulkQuery(state=> `SELECT Id FROM Patient__c WHERE Health_ID__c = '${state.data.field1}'`);
- * @example
- * bulkQuery(
- *   (state) =>
- *     `SELECT Id FROM Patient__c WHERE Health_ID__c = '${state.data.field1}'`,
- *   { pollTimeout: 10000, pollInterval: 6000 }
- * );
+ * @example <caption>Perform a bulk query on Patient records</caption>
+ * bulkQuery('SELECT Id, Name FROM Patient__c');
```
Comment (josephjclark) [inline packages/salesforce/src/Adaptor.js:324]: These examples don't use any dynamic state. We should make sure there's at least one example which uses state (or lazy state)

File: packages/salesforce/src/Adaptor.js [hunk 2]
```diff
@@ -114,12 +131,16 @@ export function describe(sObject) {
 /**
  * Retrieves a Salesforce sObject(s).
  * @public
- * @example
- * retrieve('ContentVersion', '0684K0000020Au7QAE/VersionData');
+ * @example <caption>Retrieve a specific ContentVersion record</caption>
+ * retrieve('ContentVersion', '0684K0000020Au7QAE', state => {
+ *   console.log(state.references[0]); // Log the retrieved record
```
Comment (josephjclark) [inline packages/salesforce/src/Adaptor.js:136]: Is the comment here useful? `console.log()` is  a pretty easy to understand function

File: packages/salesforce/src/Adaptor.js [hunk 3]
```diff
@@ -591,15 +616,16 @@ export function insert(sObject, attrs) {
  *
  * **The `createIf()` function has been deprecated. Use `fnIf(condition,create())` instead.**
  * @public
- * @example
- * createIf(true, 'obj_name', {
- *   attr1: "foo",
- *   attr2: "bar"
- * })
+ * @example <caption>Create a new record if `state.isNew` is true</caption>
```
Comment (josephjclark) [inline packages/salesforce/src/Adaptor.js:619]: Let's not add new examples for deprecated APIs please

**Review by josephjclark — CHANGES_REQUESTED** [review]
Yeah @mtuchi unfortunately these docs are against the wrong branch. You have examples of state.references, but we're about to remove that stuff.

This needs to be be retargeted and mapped to the epic

I've left a couple of high level comments which will need to be taken into account

**Comment by mtuchi** [comment]
Sorry @josephjclark i wasn't able to work on change request this week, I will pick this up next week. I only manage to resolved git conflicts on #691

**Comment by mtuchi** [comment]
Closing this in favor of https://github.com/OpenFn/adaptors/pull/821

**Pull request closed without merging**